### DISCUSSION TITLE
Skandika Nytta

### DISCUSSION THREAD

**Discussion by elpekos** [JaapvanEkris/openrowingmonitor#24] [Rower Profile Development]
Hello,

I'd like to install openrowingmonitor on my nytta water rower. The wheel has 1 magnet and there is 2 sensors placed at 2cm from each other. Two cables (with 2 lines) are wired to the console. How could I use them ?

Thanks

**Comment by JaapvanEkris** [comment]
I think you'd use a single sensor, and see what happens from there.

**Comment by elpekos** [comment]
ok, I'll try with only one. I'll start tomorrow and will give a feedback.

**Comment by elpekos** [comment]
Hello, 

The Nytta is a water rower. The 2 reed sensors sense the handle drive wheel. 
I've connected one sensor to GPIO 17. It works.

In the supported rower, the abilica seems to have the same sensor but I dont find any information about how to configure it.
Any ideas ?

Thanks

**Reply by Abasz** [reply]
HI, please have a look at this discussion about the two reed sensor setup: https://github.com/laberning/openrowingmonitor/discussions/95

Can you confirm that the magnet is on the handle bar sprocket spindle?

Generally the way two reed sensor monitor works by detecting the rotation of the handle bar sprocket spindle. Basically it detects the passing magnet and determines the direction (drive vs. recovery) by checking which reed is triggered first. When you pull the monitor shows you a random but consistent value for speed and distance. It cannot really know these just from the information of the handle bar (for reasons please see the [phsysics of rowing](https://github.com/JaapvanEkris/openrowingmonitor/blob/Backend_Redesign/docs/physics_openrowingmonitor.md)). For instance my monitor shows me around 30-50 watt of power even when I pull like crazy :)

If the above is you case then, as stated in the discussion there are two routes: i) you add a sensor to the impeller shaft (and magnet(s)) to detect its rotation, or ii) you settle with the random but consistent number. In the latter case, in my view, the only advantage is that you can tweak those "random but consistent" figures to something that resembles more to reality. E.g. you go to a gym test your self on a Concept 2 and you tweak ORM to reproduce something like that. It depends on you whether you deem this useful or not.

**Reply by elpekos** [reply]
Thanks for your reply. The magnet is on the first wheel. Here and you can see the 2 reed sensors.
![IMG_2173](https://github.com/JaapvanEkris/openrowingmonitor/assets/37683149/c42fe1dd-2585-4978-b68c-39d0a1ae7871)

I understand that the best solution is to add a magnet on the impellor and sense it with a hall effect sensor. Unfortunately, there is no space left on this axle... I've to find a smart solution.

Congratulations for you maintaining such a great project.

**Reply by Abasz** [reply]
To this the discussion applies.

**Comment by elpekos** [comment]
Hello,

I've designed and 3D printed a wheel with 2 magnets and a reed sensor.

![IMG_2191](https://github.com/JaapvanEkris/openrowingmonitor/assets/37683149/3eb292a6-ecba-41b5-8364-233df93b00eb)
![IMG_2192](https://github.com/JaapvanEkris/openrowingmonitor/assets/37683149/0f7f8a88-3b76-47f3-a58b-ae78cc0f1983)
![IMG_2193](https://github.com/JaapvanEkris/openrowingmonitor/assets/37683149/63168ad7-7adb-435c-bbdf-2a8134465df3)
![IMG_2194](https://github.com/JaapvanEkris/openrowingmonitor/assets/37683149/465e0aac-08be-4c9b-b137-d0c2cb7dce97)

After filtering and adjusting some parameters, I've reliable raw data : they show that the reed switch triggers 2 times for 1 magnet pass : the second time is always 1/3 of first signal. I saw the youtube video about reed sensors : it is a normal behavior when the sensor is not adjusted. How could I manage this without redesigning the sensor part ?

**Reply by Abasz** [reply]
haha I did the same for my rower, I only did for 3 magnets :)

![20230404_095734](https://github.com/JaapvanEkris/openrowingmonitor/assets/32517724/806cfc9c-c8f1-456e-b3b0-a0f7a4c063dc)
![20230404_095937](https://github.com/JaapvanEkris/openrowingmonitor/assets/32517724/d77b0bd1-ae42-49e9-adb1-62e35f251c62)

There are several discussion in this topic, lengthy once :)

https://github.com/laberning/openrowingmonitor/discussions/121 
https://github.com/laberning/openrowingmonitor/discussions/122
https://github.com/laberning/openrowingmonitor/discussions/115

What you are experiencing is de-bounce, and its normal for reed sensors. 

This is a good summary advice how to start : https://github.com/laberning/openrowingmonitor/discussions/121#discussioncomment-4868360
This is the specific answer to your question (and the my answer to the post on the link): https://github.com/laberning/openrowingmonitor/discussions/122#discussioncomment-4843636

**Reply by JaapvanEkris** [reply]
That is a beautiful setup! Nice modification.

**Comment by elpekos** [comment]
Hello, 

I switched to 4 magnets. 2 questions : 
- sprocket radius, in centimeters ?
- flywheel inertia, how to approximate ? (tank with a radius of 23cm, 10L inside)

Thanks again for your work and your support.

**Comment by Abasz** [comment]
> Hello,
> 
> I switched to 4 magnets. 2 questions :
>
> * sprocket radius, in centimeters ?

According to the docs its cm: https://github.com/laberning/openrowingmonitor/blob/v1beta/docs/rower_settings.md#review-sprocketradius-and-minumumforcebeforestroke

> * flywheel inertia, how to approximate ? (tank with a radius of 23cm, 10L inside)
flywheel inertia in water rowers are the inertia of the actual rotating wheel (the paddles that stir the water basically).

I do not think its possible to estimate the inertia without doing some measurements of this wheel. There is one discussion here: https://github.com/laberning/openrowingmonitor/discussions/113

And here you can find the conventional way of estimating inertia: https://dvernooy.github.io/projects/ergware/

Or you can adjust it until you get sensible data, assuming the rest of the settings are fine

> 
> Thanks again for your work and your support.
